ride_worksheet.rb

```diff
@@ -0,0 +1,116 @@
+Hash.new
+drivers_data = {
+  DR0001: [
+    {date: "3rd Feb 2016", 
+    rider_id: "RD0003", 
+    cost: 10, 
+    rating: 3},
+    {date: "3rd Feb 2016", 
+    rider_id: "RD0015", 
+    cost: 30, 
+    rating: 4},
+    {date: "5th Feb 2016", 
+    rider_id: "RD0003", 
+    cost: 45, 
+    rating: 1}
+  ],
+  DR0002: [
+    {date: "3rd Feb 2016",
+    rider_id: "RD0073",
+    cost: 25, 
+    rating: 5},
+    {date: "4th Feb 2016",
+    rider_id: "RD0013",
+    cost: 15,
+    rating: 1},
+    {date: "5th Feb 2016",
+    rider_id: "RD0066",
+    cost: 35, 
+    rating: 3}
+  ],
+  DR0003: [
+    {date: "4th Feb 2016",
+    rider_id: "RD0066",
+    cost: 5,
+    rating: 5},
+    {date: "5th Feb 2016",
+    rider_id: "RD0003",
+    cost: 50,
+    rating: 2}
+  ],
+  DR0004: [
+    {date: "4th Feb 2016",
+    rider_id: "RD0022",
+    cost: 5, 
+    rating: 5},
+    {date: "5th Feb 2016", 
+    rider_id: "RD0022",
+    cost: 10,
+    rating: 4},
+    {date: "5th Feb 2016",
+    rider_id: "RD0073",
+    cost: 20,
+    rating: 5}
+  ]  
+}
+
+
+
+# 1. The number of rides each driver has given 
+drivers_data.each do |driver_id, trips|
+  puts "Driver #{driver_id}'s total trips: #{trips.length}"
+end
+
+# 2. The total amount of money each driver has made
+drivers_data.each do |driver_id, trips|
+  total = 0
+  trips.each { |trip|
+  total += trip[:cost]
+}
+  puts "Driver #{driver_id}'s total earnings: $#{total}"
+  # puts "$#{total} and #{driver_id}"
+end
+
+# 3. The average rating for each driver
+drivers_data.each do |driver_id, trips|
+  ratings = 0
```

Comment:
Ratings need to use [floating point math](https://en.wikipedia.org/wiki/Floating-point_arithmetic) since we're planning to get an average and want things like 4.67 as an answer.

Because of the way Ruby converts things if you have one floating point value everything winds up floating point so if you start with `0.0` you should get the correct math.  (Though note that with floating point math you will want to `round` before you display things.)
```suggestion
  ratings = 0.0
```